test/component/login_test.dart

```diff
@@ -1,83 +1,202 @@
+import "../services/test_data/firestore_user_mock.dart";
 
 void main() {
-  setupFirebaseAuthMocks();
+  const delayNeededForAsyncFunctionExecution = Duration(seconds: 1);
+  late FakeFirebaseFirestore fakeFireStore;
+  late CollectionReference<Map<String, dynamic>> userCollection;
+  late UserRepositoryService userRepo;
 
-  final mockedProxima = ProviderScope(
-    overrides: firebaseAuthMocksOverrides,
-    child: const ProximaApp(),
-  );
+  setupFirebaseAuthMocks();
 
   setUpAll(() async {
     await Firebase.initializeApp();
   });
 
-  testWidgets("Login to Create Account Page to Home Page flow", (tester) async {
-    await tester.pumpWidget(mockedProxima);
-    await tester.pumpAndSettle();
-
-    // Check for the logo on the Login Page
-    final logoFinder = find.byKey(LoginPage.logoKey);
-    expect(logoFinder, findsOneWidget);
-
-    // Check for the slogan on the Login Page
-    final sloganFinder = find.text(LoginPage.tagLineText);
-    expect(sloganFinder, findsOneWidget);
-
-    final loginButton = find.byKey(LoginButton.loginButtonKey);
-    // Check that the login button is displayed and contains the "Login" text
-    expect(
-      find.descendant(
-        of: loginButton,
-        matching: find.text("Sign in with Google"),
-      ),
-      findsOneWidget,
-    );
-
-    await tester.tap(loginButton);
-    await tester.pumpAndSettle();
-
-    // Check that pressing login redirects to the create account page
-    final createAccountPage = find.byType(CreateAccountPage);
-    expect(createAccountPage, findsOneWidget);
-
-    // And that pushing the confirm button redirects to the home page
-    final confirmButton = find.byKey(CreateAccountPage.confirmButtonKey);
-    await tester.tap(confirmButton);
-    await tester.pumpAndSettle();
-
-    // We must now be on the home page
-    final homePage = find.byType(HomePage);
-    expect(homePage, findsOneWidget);
+  group("Existing user data in repository testing", () {
+    final expectedUser = testingUserFirestore;
+
+    setUp(() async {
+      fakeFireStore = FakeFirebaseFirestore();
+      userCollection = fakeFireStore.collection(UserFirestore.collectionName);
+      userRepo = UserRepositoryService(
+        firestore: fakeFireStore,
+      );
+      await userCollection
+          .doc(expectedUser.uid.value)
+          .set(expectedUser.data.toDbData());
+    });
+
+    ProviderScope getMockedProxima() {
+      return ProviderScope(
+        overrides: [
+          ...firebaseAuthMocksOverrides,
+          userRepositoryProvider.overrideWithValue(userRepo),
+        ],
+        child: const ProximaApp(),
+      );
+    }
+
+    testWidgets("Login and get to home page flow", (tester) async {
+      await tester.pumpWidget(getMockedProxima());
+      await tester.pumpAndSettle();
+
+      final loginButton = find.byKey(LoginButton.loginButtonKey);
+      await tester.tap(loginButton);
+      //Needs a delay to allow the existance check to complete
```

Comment:
I don't on my computer, I can remove this delay (same for all occurences below). Is this device dependent?

Reply:
It dependent on the device, on my computer the test does not pass, since we are mocking the future we can be sure that it will be executed instantly, but sometimes the future is not started yet when we return from the tap on the `loginButton`.

Reply:
Ok, that's weird. This might become a bigger issue if tests pass on some people's computer but not on other's/the CI.
Thanks :)